spec/models/category_spec.rb

```diff
@@ -1,22 +1,23 @@
 require 'rails_helper'
 
 RSpec.describe Category, type: :model do
-  before do
-    @categories = FactoryBot.create_list(:category, 4)
-    puts "*"  *  30
-    puts Category.all.inspect
-    puts "*"  *  30
-    puts Item.all.inspect
-    puts "*"  *  30
-    puts Shop.all.inspect
-    puts "*"  *  30
-    puts User.all.inspect
-    puts "*"  *  30
+  let!(:shop) { create(:shop) }
+  let!(:category_with_items) { create(:category_with_items, items_count: 2, shop: shop) }
+  let!(:categories) { create_list(:category, 3) }
+  let!(:item) { create(:item, categories: [categories[1]], shop: shop) }
+
+  it 'should create a valid instance of Category' do
+    expect(category_with_items).to be_valid
   end
 
   context 'creation' do
-    it 'needs to have a name' do
-      expect(@categories.sample.name).to be_kind_of(String)
+    it 'need to persist' do
+      expect(Category.count).to eq(4)
```

Comment:
expect { categories }.to change(Category, :count).by 4

Reply:
Est-ce qu'un expect { create_list(:category, 4) }.to change(Category, :count).by 4 fonctionne? Pcq'en utilisant le "categories",  rspec expect "change" ne détecte pas de changement parce que l'opération de création des catégories se fait en amont avec le "let" si j'ai bien compris. Et le test ne passe pas sinon du coup :s